projects/gui/src/tournamentsettingswidget.cpp

```diff
@@ -32,6 +32,73 @@ TournamentSettingsWidget::TournamentSettingsWidget(QWidget *parent)
 		ui->m_seedsSpin->setEnabled(checked);
 	});
 
+	// number of rounds limits repeats
+	connect(ui->m_roundsSpin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+		[=](int value)
+	{
+		if (!ui->m_gauntletRadio->isChecked())
+			value = 1;
+		int limit = value * ui->m_gamesPerEncounterSpin->value();
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// number of games per encounter limits repeats
+	connect(ui->m_gamesPerEncounterSpin,
+		static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), [=](int value)
+	{	int rounds = ui->m_gauntletRadio->isChecked() ?  ui->m_roundsSpin->value() : 1;
+		int limit = value * rounds;
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// update repeats after rounds
+	connect(ui->m_roundsSpin,
+		static_cast<void (QSpinBox::*)()>(&QAbstractSpinBox::editingFinished), [=]()
+	{
+		int value = ui->m_repeatSpin->value();
+		ui->m_repeatSpin->valueChanged(value);
+	});
+
+	// update repeats after games per encounter
+	connect(ui->m_gamesPerEncounterSpin,
+		static_cast<void (QSpinBox::*)()>(&QAbstractSpinBox::editingFinished),[=]()
+	{
+		int value = ui->m_repeatSpin->value();
+		ui->m_repeatSpin->valueChanged(value);
+	});
+
+	// update repeats via rounds spin after tournament type changed
+	connect(ui->m_gauntletRadio,
+		static_cast<void (QRadioButton::*)(bool)>(&QAbstractButton::toggled),[=]()
+	{
+		int value = ui->m_roundsSpin->value();
+		ui->m_roundsSpin->valueChanged(value);
+	});
+
+	// repeats dependent on games per encounter, rounds and tournament type
+	connect(ui->m_repeatSpin,static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+		[=](int valueIn)
+	{
+		static int store = 1;
+		int value = valueIn;
+		int step = (ui->m_repeatSpin->value() > store) ? 1 : -1;
+		int games = ui->m_gamesPerEncounterSpin->value();
+		int rounds = 1;
+		if (ui->m_gauntletRadio->isChecked())
+			rounds = ui->m_roundsSpin->value();
+
+		while (games % value
+		&&     (value % games || (games * rounds) % value) )
+			value += step;
+
+		ui->m_repeatSpin->setValue(value);
```

Comment:
I thinks it's quite questionable to call `setValue` from `valueChanged`; there's a risk of infinite loop.

Reply:
This is indeed a recursion, but the call depth will not exceed 2.

Reply:
I have just simplified this, so the setter is only called if needed.

Reply:
Reworked.